src/sexpr.rs

```diff
@@ -412,6 +477,23 @@ impl<'a> Lexer<'a> {
 
         &self.chars[start..end]
     }
+
+    /// Scan a string literal. `start` is expected to be the offset of the opening `"`. The scanned
+    /// string excludes both the start and end quotes.
+    fn scan_string(&mut self, start: usize) -> &'a str {
+        while let Some((ix, c)) = self.indices.next() {
+            if c == '\\' {
+                self.indices.next();
+                continue;
+            }
+
+            if c == '"' {
+                return &self.chars[start + 1..ix];
+            }
+        }
+
+        "\"\""
```

Comment:
I'm not totally sure this is the right behavior. The lexer currently can't signal an error, so this will instead say that an unterminated string literal is just the empty string.

Reply:
This is a little bit hacky indeed. I think the two best options, depending on your taste, are either

1. Make the `Iterator for Lexer` impl yield items of type `Result<Token<'a>, LexerError>` and make sure it is a fused adapter such that after it returns an `Err(_)` it always returns `None` afterwards.
2. Add a `Token::Error` variant and then translate that into a proper error during parsing.

What do you think?

Reply:
Option 2 seems easy to add in. Parsing currently returns an `Option`, so I'll change that to a `Result` while I'm here.